include/envoy/grpc/async_client.h

```diff
@@ -91,6 +104,20 @@ class AsyncRequestCallbacks {
                          Tracing::Span& span) PURE;
 };
 
+class AsyncRequestCallbacks : public RawAsyncRequestCallbacks {
+public:
+  virtual ~AsyncRequestCallbacks() {}
+
+  void onSuccessRaw(Buffer::InstancePtr response, Tracing::Span& span) override;
+  /**
+   * Called when the async gRPC request succeeds. No further callbacks will be invoked.
+   * @param response the gRPC response.
+   * @param span a tracing span to fill with extra tags.
+   * NB: requires overriding createEmptyResponse().
+   */
+  virtual void onSuccessUntyped(ProtobufTypes::MessagePtr&& response, Tracing::Span& span) PURE;
```

Comment:
I think if we have in the "this is the interface for a gRPC implementation" only raw callbacks/methods, and then in `source/common/grpc` we can add an lightweight proto -> raw wrapper layer, this could simplify the interface.

Reply:
I struggled a bit with the best way to do this, and decided to follow the example of *Untyped here.  I am not sure how your suggestion would work out for the common case since typed messages are likely to be used fairly commonly.

Perhaps we can talk offline and you can give me a better idea of what you are proposing here.

Reply:
Done.